phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -75,6 +85,38 @@ public Void run() throws Exception {
         }
     }
 
+    private static List<Mutation> getMutationsForSystemTaskTable(
+            PhoenixConnection conn, PreparedStatement stmt,
+            boolean accessCheckEnabled) throws IOException {
+        // we need to mutate SYSTEM.TASK with HBase/login user if access is enabled.
+        if (accessCheckEnabled) {
+            return User.runAsLoginUser(() -> {
+                final RpcCall rpcContext = RpcUtil.getRpcContext();
+                // setting RPC context as null so that user can be reset
+                try {
+                    RpcUtil.setRpcContext(null);
+                    stmt.execute();
+                    // retrieve mutations for SYSTEM.TASK upsert query
+                    return conn.getMutationState().toMutations().next()
```

Comment:
If other mutations were made using the same connection, we will be returning the joined state of all of them here right? Don't we want to restrict to just returning mutations corresponding to the upsert into the SYSTEM.TASK table?